### PR TITLE
<fix>(request): Fix unexpected CookieError during cookie parsing

### PR DESCRIPTION
**Pull request by yasek**
Fix unexpected CookieError by splitting Cookie header into parts and parsing them independently.

### CODE REVIEW

**Comment by yasek** [comment]
@kgriffs Also, is it possible to include this fix into current release?

File: falcon/request.py [hunk 1]
```diff
@@ -896,7 +896,13 @@ def cookies(self):
             # NOTE(tbug): We might want to look into parsing
             # cookies ourselves. The SimpleCookie is doing a
             # lot if stuff only required to SEND cookies.
-            parser = SimpleCookie(self.get_header('Cookie'))
+            cookie_header = self.get_header('Cookie', default='')
```
Comment (kgriffs) [inline falcon/request.py]: test: Could you add a test for the case when the `Cookie` header is missing?

File: falcon/request.py [hunk 2]
```diff
@@ -896,7 +896,13 @@ def cookies(self):
...
+            parser = SimpleCookie()
+            for cookie_part in cookie_header.split('; '):
+                try:
+                    parser.load(cookie_part)
```
Comment (kgriffs) [inline falcon/request.py]: fix later: Calling `load()` multiple times will result in a performance degradation. I'll create an issue to address this in 1.4.

File: falcon/request.py (hunk 1 above)
Comment (yasek) [inline falcon/request.py]: Fixed

**Pull request merged**